base/stream.jl

```diff
@@ -929,9 +935,11 @@ function read!(s::LibuvStream, a::Array{UInt8, 1})
             write(newbuf, sbuf)
             wait_readnb(s, nb)
         finally
+            nba = nb_available(s.buffer)
             s.buffer = sbuf
+            nba < nb && resize!(a, nba)
             if !isempty(s.readnotify.waitq)
-                start_reading(x) # resume reading iff there are currently other read clients of the stream
+                start_reading(s) # resume reading iff there are currently other read clients of the stream
```

Comment:
I think no one having ever seen a UndefVarError or suchlike from here suggests that this branch never executes.

I also suspect that after the stream's buffer has been switched, all other simultaneous pended reads that are waiting the readnotify event in wait_readnb() loop begin checking the loop condition ```while isopen(x) && nb_available(x.buffer) < nb``` with the substituted buffer. But when they return from the wait_readnb() and strat copying data (i.e. ```read!(sbuf, a)```) they do it from their saved local reference ```sbuf``` but not from the actual stream's buffer.